### PR TITLE
Fix some places where python2 packages fail to evaluate.

### PR DESCRIPTION
**Pull request by tomprince**
I unfortunately still depend on some python2 packages. It appears
that these two packages fail to evaluate, in a way that `tryEval`
doesn't catch. This changes them to be explicitly disabled there.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python2-packages.nix [hunk 1]
```diff
@@ -12,7 +12,7 @@ with self; with super; {
     inherit (pkgs) antlr4;
   };
 
-  appleseed = toPythonModule (pkgs.appleseed.override { inherit (self) python; });
+  appleseed = disabled super.appleseed;
```
Comment (SuperSandro2000) [inline pkgs/top-level/python2-packages.nix:15]: If it is always disabled we can also remove it.

**Comment by tomprince** [comment]
It is only disabled on py2, but the py2 set of packages inherits from the
py3 set, so we need to override the inherited packages to disable them.

File: pkgs/top-level/python2-packages.nix (hunk 1 above)
Comment (tomprince) [reply pkgs/top-level/python2-packages.nix:15]: It is *not* disabled on python3. This is (as far as I can tell) the way it was suggested to me that I disable a package only on python2 in  #148329 (based on the existing examples in this file).

Comment (SuperSandro2000) [reply pkgs/top-level/python2-packages.nix:15]: This should be done via disabled in the package itself. This file should only contain python2 only packages or entries when we load older versions via an other file.

Comment (jonringer) [reply pkgs/top-level/python2-packages.nix:15]: For python2, I think this is fine

Comment (jonringer) [reply pkgs/top-level/python2-packages.nix:15]: we are trying to get rid of it for 22.05

**Review by jonringer — APPROVED** [review]
LGTM

failures are broken on target branch
```
https://github.com/NixOS/nixpkgs/pull/148328

2 packages failed to build:
python38Packages.appleseed python39Packages.appleseed
```

**Pull request merged**